bundle/jsky.app.ot/src/main/scala/jsky/app/ot/tpe/feat/TpeEphemerisFeature.scala

```diff
@@ -27,22 +31,65 @@ class TpeEphemerisFeature extends TpeImageFeature("Ephemeris", "Show interpolate
       p
     }.toOption
 
-  def getScreenEphemeris: ScreenEphemeris =
+  def toScreenEphemeris(e: Ephemeris): ScreenEphemeris =
+    e.toList.flatMap { case (t, c) => toScreenCoordinates(c).strengthL(t) }
+
+  def getEphemeris: Ephemeris =
     getContext.targets.base
       .flatMap(_.getNonSiderealTarget)
-      .fold(Ephemeris.empty)(_.ephemeris).toList
-      .flatMap { case (t, c) => toScreenCoordinates(c).strengthL(t) }
+      .fold(Ephemeris.empty)(_.ephemeris)
+
+  def getScreenEphemeris: ScreenEphemeris =
+    toScreenEphemeris(getEphemeris)
+
+  def obsTime: Option[Long] =
+    getContext.schedulingBlock.flatMap(_.duration) orElse
+    getContext.obsShell.map(remainingTime)
+
+  def obsScreenEphemeris: Option[ScreenEphemeris] =
+    for {
+      lo <- getContext.schedulingBlockStart
+      hi <- obsTime.map(_ + lo)
+      e  <- getEphemeris.iSlice(lo, hi)
+    } yield toScreenEphemeris(e)
+
+  def draw(g: Graphics, tii: TpeImageInfo): Unit =
+    draw2D(g.asInstanceOf[Graphics2D])
+
+  def draw2D(g: Graphics2D): Unit = {
+
+    // This seems to help
+    g.setRenderingHint(KEY_ANTIALIASING,      VALUE_ANTIALIAS_ON)
```

Comment:
It would be good to do a quick check on Linux/Windows as these keys maybe system dependent. In particular a check on some of the workstations running CentOS